cmd/deploy.go

```diff
@@ -144,6 +144,13 @@ func runDeploy(commandContext *cmdctx.CmdContext) error {
 	} else {
 		// no image specified, build one
 		buildArgs := map[string]string{}
+
+		if commandContext.AppConfig.Build.Args != nil {
```

Comment:
You might need to check that `commandContext.AppConfig.Build` isn't nil first. Try it out with an app without a `[build]` section

Reply:
Ran it without a [build] and it worked, will add in the check tho. I suspect that the toml library fills in for ancestors... ie, build.args infers build as a section. So if there's a build.x, there will be an build to hold the .x in the tree. Our marshall to Toml code ends up putting in a build too which gets added to the .toml file.

So I think it's ok.

Reply:
Well, I was so wrong there. Blows up when there's no build or build subsections at all. So... fix incoming.